spec/requests/api/v1/users/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V1::Users::OmniauthCallbacksController, type: :request do
+  describe 'POST #google_oauth2' do
+    context 'given the email is available' do
+      it 'returns a successful response' do
+        application = Fabricate(:application)
+        allow(Doorkeeper::Application).to receive(:find_by).and_return(application)
+        OmniAuth.config.test_mode = true
+        OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new(
+          {
+            provider: :google_oauth2,
+            uid: '123545',
+            info: { email: 'email@email.com' }
+          }
+        )
+
+        post '/api/v1/users/auth/google_oauth2/callback'
+
+        expect(response).to have_http_status(:success)
+      end
+    end
+
+    context 'given the email is not available' do
+      it 'returns an unprocessable entity response with errors' do
```

Comment:
```suggestion
      it 'returns unprocessable_entity error' do
```

Just to be consistent with the other tests ;-)